lib/functions.ts

```diff
@@ -1,9 +1,7 @@
 /**
- * Checks if the tagName string value matches
- * h1, h2, h3, h4, h5, h6 and returns boolean
- * @param tagName {String}
+ * Checks if the given `node` is an instanceof HTMLHeadingElemnt
  * @returns {boolean}
  */
-export function isHeader(tagName: string): boolean {
-  return !!tagName.match(/^H[1-6]$/i);
+export function isHeadingElement(node: Node | null): node is HTMLHeadingElement {
```

Comment:
[TIL](https://developer.mozilla.org/en-US/docs/Web/API/HTMLHeadingElement) thanks!

I didn't know this class exists. If you want, you can remove this and just inline the instanceof check, as far as TypeScript is concerned, it's the same thing. You can keep this predicate here if you prefer, too.